### PR TITLE
Rework error handling and move log data handling

### PR DESCRIPTION
**Pull request by mramendi**

### CODE REVIEW

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-errors-con.adoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+* `ExecutionErrorManager`: The entry point for error handling. This class is integrated with `RuntimeManager`, which is then responsible for providing it to the underlying `KieSession` and `TaskService`. It provides access to other classes in the execution error handling mechanism.
++
+When the {PROCESS_ENGINE} creates a `RuntimeManager` instance, it also creates a corresponding `ExecutionErrorManager` instance.
++
+* `ExecutionErrorHandler`: The primary class for error handling. This class is implemented in the {PROCESS_ENGINE} and you normally do not need to customize or extend it directly. `ExecutionErrorHandler` calls error filters to process particular errors and calls `ExecutionErrorStorage` to store error information.
++
+The `ExecutionErrorHandler` is bound to the life cycle of `RuntimeEngine`; it is created when a new runtime engine is created, and is destroyed when `RuntimeEngine` is disposed. A single instance of the `ExecutionErrorHandler` is used within a given execution context or transaction. Both `KieSession` and `TaskService` use that instance to inform the error handling about processed nodes/tasks. `ExecutionErrorHandler` is informed about the following events:
```
Comment (hmanwani-rh) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-errors-con.adoc:21]:
```suggestion
The `ExecutionErrorHandler` is bound to the life cycle of `RuntimeEngine`; it is created when a new runtime engine is created and is destroyed when `RuntimeEngine` is disposed. A single instance of the `ExecutionErrorHandler` is used within a given execution context or transaction. Both `KieSession` and `TaskService` use that instance to inform the error handling about processed nodes or tasks. `ExecutionErrorHandler` is informed about the following events:
```

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-errors-con.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
++
+** Starting of processing of a given node instance.
```
Comment (hmanwani-rh) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-errors-con.adoc:23]: As per IBM SG If a list contains incomplete sentences, then end punctuation should not be added. 
```suggestion
** Starting of processing of a given node instance
```

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-errors-con.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+** Completion of processing of a given node instance.
```
Comment (hmanwani-rh) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-errors-con.adoc:24]:
```suggestion
** Completion of processing of a given node instance
```

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-errors-con.adoc [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+** Starting of processing of a given task instance.
```
Comment (hmanwani-rh) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-errors-con.adoc:25]:
```suggestion
** Starting of processing of a given task instance
```

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-errors-con.adoc [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+** Completion of processing of a given task instance.
```
Comment (hmanwani-rh) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-errors-con.adoc:26]:
```suggestion
** Completion of processing of a given task instance
```

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-errors-con.adoc [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
++
+* `ExecutionErrorStorage`: The pluggable storage class for execution error information.
++
+When the {PROCESS_ENGINE} creates a `RuntimeManager` instance, it also creates a corresponding `ExecutionErrorStorage` instance. Then the `ExecutionErrorHandler` class calls this `ExecutionErrorStorage` instance to store information abiout every execution error.
++
+The default storage implementation uses a database table to store all the available information for every error. Different detail levels might be available for different error types, as some errors might not permit extraction of detailed information.
++
+* A number of _filters_ that process particular types of execution errors. You can add custom filters.
+
+Every execution error is, by default, recorded as _unacknowledged_. You can use {CENTRAL} to view all recorded execution errors and to acknowledge them. You can also create jobs that automatically acknowledge all or some execution errors.
```
Comment (hmanwani-rh) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-errors-con.adoc:38]:
```suggestion
By default, every execution error is recorded as _unacknowledged_. You can use {CENTRAL} to view all recorded execution errors and to acknowledge them. You can also create jobs that automatically acknowledge all or some execution errors.
```

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-error-types-and-filters-ref.adoc [hunk 7]
```diff
@@ -0,0 +1,39 @@
+[id='execution-error-types-and-filters-ref']
+= Execution error types and filters
+
+Execution error handling attempts to catch and handle any kind of error. However, users might need to handle different errors in different ways. Also, different detailed information is available for different types of errors.
+
+The error handling mechanism supports pluggable _filters_. Every filter processes a particular type of error. You can add filters that process specific errors in different ways, overriding default processing.
+
+A filter is an implementation of the `ExecutionErrorFilter` interface. This interface builds instances of `ExecutionError`, which are later stored using the `ExecutionErrorStorage` class.
+
+The `ExecutionErrorFilter` interface has following methods:
```
Comment (hmanwani-rh) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-error-types-and-filters-ref.adoc:10]:
```suggestion
The `ExecutionErrorFilter` interface has the following methods:
```

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-error-types-and-filters-ref.adoc [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+* `accept`: indicates if an error can be processed by the filter.
```
Comment (hmanwani-rh) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-error-types-and-filters-ref.adoc:12]:
```suggestion
* `accept`: Indicates if an error can be processed by the filter
```

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-error-types-and-filters-ref.adoc [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+* `filter`: processes an error and returns the `ExecutionError` instance
```
Comment (hmanwani-rh) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-error-types-and-filters-ref.adoc:13]:
```suggestion
* `filter`: Processes an error and returns the `ExecutionError` instance
```

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-error-types-and-filters-ref.adoc [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+* `getPriority`: indicates the priority for this filter
```
Comment (hmanwani-rh) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-error-types-and-filters-ref.adoc:14]:
```suggestion
* `getPriority`: Indicates the priority for this filter
```

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-error-types-and-filters-ref.adoc [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+The execution error handler processes each error separately. For each error, it starts calling the `accept` method of all registered filters, starting with the filters that have a lower priority value. If the `accept` method of a filter returns `true`, the handler calls the `filter` method of this filter and does not call any other filters.
```
Comment (hmanwani-rh) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-error-types-and-filters-ref.adoc:16]:
```suggestion
The execution error handler processes each error separately. For each error, it starts calling the `accept` method of all registered filters, starting with the filters that have a lower priority value. If the `accept` method of a filter returns `true`, the handler calls the `filter` method of the filter and does not call any other filters.
```

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-error-types-and-filters-ref.adoc [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+Because of the priority system, only one filter processes any error. More specialized filters have lower priority values. An error that was not accepted by one of the specialized filters reaches generic filters that have higher priority values.
```
Comment (hmanwani-rh) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/execution-error-types-and-filters-ref.adoc:18]:
```suggestion
Due to the priority system, only one filter processes any error. More specialized filters have lower priority values. An error that is not accepted by any specialized filters reaches generic filters that have higher priority values.
```

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/error-list-cleanup-proc.adoc [hunk 13]
```diff
@@ -0,0 +1,27 @@
+[id='error-list-cleanup-proc']
+= Cleaning up the error list
+
+The {PROCESS_ENGINE} stores execution errors in the `ExecutionErrorInfo` database table. If you want to clean up the table and remove errors permanently, you can schedule a job with the `org.jbpm.executor.commands.ExecutionErrorCleanupCommand` command.
```
Comment (emmurphy1) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/error-list-cleanup-proc.adoc:4]:
```suggestion
The {PROCESS_ENGINE} stores execution errors in the `ExecutionErrorInfo` database table. If you want to clean up the table and remove errors permanently, you can use the `org.jbpm.executor.commands.ExecutionErrorCleanupCommand` command to  schedule a job.
```

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/autoacknowledge-execution-errors-proc.adoc [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+
+.Procedure
+
+. In {CENTRAL}, click *Menu* -> *Manage* -> *Jobs*.
+. In the top right of the screen, click *New Job*.
+. Type any identifier for the job into the *Business Key* field.
+. In the *Type* field, enter the type of the auto-acknowledge job:
+** `org.jbpm.executor.commands.error.JobAutoAckErrorCommand`: Acknowledge all execution errors of type `Job` where the job to which the error relates is now cancelled, completed, or rescheduled for another execution.
+** `org.jbpm.executor.commands.error.TaskAutoAckErrorCommand`: Acknowledge all execution errors of type `Task` where the task to which the error relates is in an exit state (completed, failed, exited, obsolete).
+** `org.jbpm.executor.commands.error.ProcessAutoAckErrorCommand`: Acknowledge all execution errors of any type where the process instance from which the error originates is already finished (completed or aborted), or the task from which the error originates is already finished.
```
Comment (emmurphy1) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/autoacknowledge-execution-errors-proc.adoc:19]: abort verb
Use only if the term is part of the existing product terminology; for example, a command option or a button
label. In all other instances, replace with "cancel", "stop", "end", or "fail".

Search for in doc.

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/error-list-cleanup-proc.adoc [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+
+The command deletes execution errors that are associated with completed or aborted process instances.
+
+.Procedure
+
+. In {CENTRAL}, click *Menu* -> *Manage* -> *Jobs*.
```
Comment (emmurphy1) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/error-list-cleanup-proc.adoc:10]: As before

File: doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/manage-execution-errors-proc.adoc [hunk 16]
```diff
@@ -0,0 +1,14 @@
+[id='manage-execution-errors-proc-{context}']
+= Managing execution errors
+
+By definition, every process error that is detected and stored is unacknowledged and must be handled by someone or something (in case of automatic error recovery). You can view a filtered list of errors that were or were not acknowledged. Acknowledging an error saves the user information and time stamp for traceability.
+
+.Procedure
+. In {CENTRAL}, go to *Menu* -> *Manage* -> *Execution Errors*.
```
Comment (emmurphy1) [inline doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/manage-execution-errors-proc.adoc:7]:
```suggestion
. In {CENTRAL}, select *Menu* -> *Manage* -> *Execution Errors*.
```

Comment (emmurphy1) [reply doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/manage-execution-errors-proc.adoc:7]: I l iike to say what happens after you that. So.. The Manage page appears.

Now you have context for the step.

Comment (mramendi) [reply doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/manage-execution-errors-proc.adoc:7]: Probably better but I think we need to be consistent, and do this for a lot of steps in many parts of this same document (including subsequent steps here). Afraid I can't pull that off right now. Something to do in quality improvement - perhaps same run as "clarify optional steps"

**Pull request merged**